### PR TITLE
Add support for lowering linalg_ext.sort to loops on memrefs.

### PR DESCRIPTION
**Pull request by hanhanW**
This is a step towards https://github.com/google/iree/issues/6154

### CODE REVIEW

**Comment by hanhanW** [comment]
This depends on https://github.com/google/iree/pull/6303

File: iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
+// Copyright 2021 The IREE Authors
```
Comment (MaheshRavishankar) [inline iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp:1]: Rename to "ConvertToLoops.cpp"?

File: iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+struct BubbleSortConversion : public OpRewritePattern<linalg_ext::SortOp> {
+  using OpRewritePattern<linalg_ext::SortOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg_ext::SortOp op,
+                                PatternRewriter& rewriter) const final {
+    if (!op.hasBufferSemantics()) return failure();
+    // This can be relaxed if the inputs and the outputs are not tied. Will need
+    // a copy op.
+    if (!llvm::all_of(llvm::zip(op.getInputOperands(), op.getOutputOperands()),
```
Comment (MaheshRavishankar) [inline iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp:34]: Interesting requirement. This seems off, but I see why you have this here. Might have to revisit the `sort` op specification and how the `tensor` and `memref` versions are described. For now this is OK.

File: iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  void getDependentDialects(DialectRegistry& registry) const override {
+    registry
+        .insert<StandardOpsDialect, memref::MemRefDialect, scf::SCFDialect>();
+  }
+
+  void runOnOperation() override {
+    OwningRewritePatternList patterns(&getContext());
+    MLIRContext* context = &getContext();
+
+    // clang-format off
```
Comment (MaheshRavishankar) [inline iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp:141]: Nit: usually I keep clang-format on. Just easier.

**Review by MaheshRavishankar — APPROVED** [review]

File: iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp (hunk 1 above)
Comment (hanhanW) [reply iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp]: Done. I actually prefer `ConvertToLoops`, but somehow I was following the naming in MLIR core.

File: iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp (hunk 2 above)
Comment (hanhanW) [reply iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp]: As offline discussion, the op should not take inputs. The op definition is updated, so we no longer need the check.

File: iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp (hunk 3 above)
Comment (hanhanW) [reply iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp]: done

**Pull request merged**